lib/zk_promise.js

```diff
@@ -166,16 +166,12 @@ class ZooKeeperPromise extends ZooKeeper {
      */
     promisify(fn, args) {
         return new ZkPromise((resolve, reject) => {
-            const callback = (rc, error, result) => {
-                if (rc) {
-                    reject(rc);
-                } else if (args.length > 3) {
-                    // if there are multiple success values, we return an array
-                    Array.prototype.shift.call(args, 1);
-                    Array.prototype.shift.call(args, 1);
-                    resolve(args);
+            const callback = (rc, error, ...result) => {
+                if (rc !== 0) {
+                    reject(new Error(`${rc} ${error}`));
                 } else {
-                    resolve(result);
+                    const toReturn = result.length === 1 ? result[0] : result;
```

Comment:
After a good night of sleep I have second thoughts about the return values here. I doubt the promisified version of ever (before the ES 2017 conversion) has worked correctly when the client returns two values. So, instead I think the return values should be a "tuple": something like {children, stats} instead of an array.